src/stactools/naip/utils.py

```diff
@@ -59,3 +68,24 @@ def _parse(lines, group_indent=0):
             return result
 
     return _parse(md_text.split("\n"), group_indent=0)["Metadata"]
+
+
+def missing_element(attribute: str) -> Callable[[str], Exception]:
+    def get_exception(xpath: str) -> Exception:
+        return MissingElement(
+            f"Could not find attribute `{attribute}` at xpath '{xpath}'."
+        )
+
+    return get_exception
+
+
+def get_id_date(cog_href) -> Tuple[str, datetime]:
+    resource_desc = os.path.basename(cog_href)
+    name = os.path.splitext(resource_desc)[0]
+    pattern = r"(m)_(\d{7})_(\w{2})_(\d{2})_(\d{1})_(\d{8})"
```

Comment:
this should be extracted to a compiled regex at the top-level, e.g.

NAIP_FILENAME_REGEX = re.compile(r"(m)_(\d{7})_(\w{2})_(\d{2})_(\d{1})_(\d{8})")

and than can be used like:

NAIP_FILENAME_REGEX.search(name)